Clarify API support in waypoint (#15949)
istio/istio.io: content/en/docs/ambient/usage/l7-features/index.md

By adding a waypoint proxy to your traffic flow you can enable more of [Istio's features](/docs/concepts). Waypoints are configured using the {{< gloss "gateway api" >}}Kubernetes Gateway API{{< /gloss >}}.

{{< warning >}}

> Usage of VirtualService remains at Alpha when used with the ambient data plane mode, and mixing with Gateway API configuration is  not supported, and will lead to undefined behavior.

Reviewer: ```suggestion
Usage of VirtualService with the ambient data plane mode is considered Alpha. Mixing with Gateway API configuration is not supported, and will lead to undefined behavior.
```
